gammapy/estimators/flux.py

```diff
@@ -113,16 +113,16 @@ def get_scale_model(self, models):
         ref_model = models[self.source].spectral_model
         scale_model = ScaleSpectralModel(ref_model)
 
-        for scaled_parameter in ref_model.parameters:
-            if scaled_parameter.is_norm:
-                break
-        else:
+        norms = Parameters([p for p in ref_model.parameters if p.is_norm])
+        if len(norms) == 0 or len(norms.free_parameters)>1 :
```

Comment:
This still pick the first norm when all parameters are frozen as it should be the case if there only one norm to be backward compatible. But we could also raise if (len(norms>1) and len(norms.free_parameters)==0).